src/mavlink-router/binlog.cpp

```diff
@@ -75,22 +75,31 @@ int BinLog::write_msg(const struct buffer *buffer)
 {
     const bool mavlink2 = buffer->data[0] == MAVLINK_STX;
     uint32_t msg_id;
+    uint8_t *payload;
     uint16_t payload_len;
     uint8_t trimmed_zeros;
+    uint8_t source_system_id;
     mavlink_remote_log_data_block_t *binlog_data;
 
     if (mavlink2) {
         struct mavlink_router_mavlink2_header *msg
             = (struct mavlink_router_mavlink2_header *)buffer->data;
         msg_id = msg->msgid;
+        payload = buffer->data + sizeof(struct mavlink_router_mavlink2_header);
         payload_len = msg->payload_len;
+        source_system_id = msg->sysid;
     } else {
         struct mavlink_router_mavlink1_header *msg
             = (struct mavlink_router_mavlink1_header *)buffer->data;
         msg_id = msg->msgid;
+        payload = buffer->data + sizeof(struct mavlink_router_mavlink1_header);
         payload_len = msg->payload_len;
+        source_system_id = msg->sysid;
     }
 
+    /* Check if we should start or stop logging */
+    _handle_auto_start_stop(msg_id, source_system_id, payload);
```

Comment:
I don't like how this is a special case and the routing to the logging endpoint is plugged in the middle of this function. What's the problem with letting it inside the logendpoint itself implementing the normal pattern like other endpoints with accept_msg() + write()?

Reply:
I agree it's not that nice. But from what I see `accept_msg` cannot filter by `source_system_id` which is what we want here.